locallib.php

```diff
@@ -903,46 +903,39 @@ public function take_from_student($mformdata) {
         return true;
     }
 
-    public function take_from_form_data($formdata) {
+    public function take_attendance($formdata) {
         global $DB, $USER;
-        // TODO: WARNING - $formdata is unclean - comes from direct $_POST - ideally needs a rewrite but we do some cleaning below.
+
+        if (isset($formdata->takenby) && !empty($formdata->takenby)) {
```

Comment:
This makes me a little uncomfortable, it would be better if we could pass $takenby as a param in the function and use $USER->id if it is blank rather than use it within $formdata (the use of $formdata really needs a rewrite anyway)